File: angular-ui/src/app/models/llm-usage-summary.ts
export interface LlmUsageByModel {
  model: string;
  provider: string;
  inputTokens: number;
  outputTokens: number;
  totalTokens: number;
  requestCount: number;
}

export interface LlmUsageDailyBreakdown {
  date: string;
  codesystem: string;
  model: string;
  provider: string;
  inputTokens: number;
  outputTokens: number;
  totalTokens: number;
  requestCount: number;
}

export interface LlmUsageSummary {
  period: string;
  codesystem?: string;
  model?: string;
  startDate?: string;
  endDate?: string;
  inputTokens: number;
  outputTokens: number;
  totalTokens: number;
  requestCount: number;
  byModel: LlmUsageByModel[];
  dailyBreakdown: LlmUsageDailyBreakdown[];
}

export type LlmUsagePeriod = 'day' | 'week' | 'month' | '3months' | '6months' | 'year' | 'all';

export const LLM_USAGE_PERIODS: { value: LlmUsagePeriod; label: string }[] = [
  { value: 'day', label: 'Last day' },
  { value: 'week', label: 'Week' },
  { value: 'month', label: 'Month' },
  { value: '3months', label: '3 months' },
  { value: '6months', label: '6 months' },
  { value: 'year', label: 'Year' },
  { value: 'all', label: 'All time' }
];

File: api/src/main/java/org/snomed/simplex/rest/pojos/LlmUsageSummary.java
package org.snomed.simplex.rest.pojos;

import java.util.ArrayList;
import java.util.List;

public class LlmUsageSummary {

	private String period;
	private String codesystem;
	private String model;
	private String startDate;
	private String endDate;
	private long inputTokens;
	private long outputTokens;
	private long totalTokens;
	private long requestCount;
	private long conceptsTranslated;
	private List<LlmUsageByModel> byModel = new ArrayList<>();
	private List<LlmUsageDailyBreakdown> dailyBreakdown = new ArrayList<>();

	public String getPeriod() {
		return period;
	}

	public void setPeriod(String period) {
		this.period = period;
	}

	public String getCodesystem() {
		return codesystem;
	}

	public void setCodesystem(String codesystem) {
		this.codesystem = codesystem;
	}

	public String getModel() {
		return model;
	}

	public void setModel(String model) {
		this.model = model;
	}

	public String getStartDate() {
		return startDate;
	}

	public void setStartDate(String startDate) {
		this.startDate = startDate;
	}

	public String getEndDate() {
		return endDate;
	}

	public void setEndDate(String endDate) {
		this.endDate = endDate;
	}

	public long getInputTokens() {
		return inputTokens;
	}

	public void setInputTokens(long inputTokens) {
		this.inputTokens = inputTokens;
	}

	public long getOutputTokens() {
		return outputTokens;
	}

	public void setOutputTokens(long outputTokens) {
		this.outputTokens = outputTokens;
	}

	public long getTotalTokens() {
		return totalTokens;
	}

	public void setTotalTokens(long totalTokens) {
		this.totalTokens = totalTokens;
	}

	public long getRequestCount() {
		return requestCount;
	}

	public void setRequestCount(long requestCount) {
		this.requestCount = requestCount;
	}

	public long getConceptsTranslated() {
		return conceptsTranslated;
	}

	public void setConceptsTranslated(long conceptsTranslated) {
		this.conceptsTranslated = conceptsTranslated;
	}

	public List<LlmUsageByModel> getByModel() {
		return byModel;
	}

	public void setByModel(List<LlmUsageByModel> byModel) {
		this.byModel = byModel;
	}

	public List<LlmUsageDailyBreakdown> getDailyBreakdown() {
		return dailyBreakdown;
	}

	public void setDailyBreakdown(List<LlmUsageDailyBreakdown> dailyBreakdown) {
		this.dailyBreakdown = dailyBreakdown;
	}
}
